ui-lib/src/main/java/co/electriccoin/zcash/ui/MainActivity.kt

```diff
@@ -133,6 +134,35 @@ class MainActivity : ComponentActivity() {
         }
     }
 
+    /**
+     * The body of this function is meat to be execute only while the app run on Firebase Test Lab or
+     * Google Play pre-launch report tests.
+     *
+     * It skips creating a new or restoring an existing wallet by persisting a wallet with a mock seed.
+     * As a result of this, it navigates the monkey test runner from onboarding screens right to the
+     * app home screen.
+     *
+     * @return true if the app is currently being messed with by a monkey test runner and the described
+     * has proceeded or false otherwise.
+     */
+    private fun changeAppFlowForAutomationTests(): Boolean {
```

Comment:
This method is doing a boolean check but also making asynchronous state changes that don't finish before the method returns (due to the call to the ViewModel).  I have a concern that callers might not fully realize that from just the method signature.  I've been trying to think how to make that more clear.

One thought is splitting into two methods.  These are terrible names, just examples:
 - isBypassOnboardingForRoboTest()
 - startOnboardingBypassForRoboTest()

There might be other ways too.

Reply:
Yes, understand. I've updated the code structure to be more clear, imho. See the last commit, please.

Reply:
This is a nice improvement!